contractcourt/htlc_timeout_resolver.go

```diff
@@ -551,21 +532,15 @@ func (h *htlcTimeoutResolver) sweepSecondLevelTx() error {
 // resolveSecondLevelTxLegacy sends a second level timeout transaction to the
 // utxo nursery. This transaction uses the legacy SIGHASH_ALL flag.
 func (h *htlcTimeoutResolver) resolveSecondLevelTxLegacy() error {
-	log.Debugf("%T(%v): incubating htlc output", h,
-		h.htlcResolution.ClaimOutpoint)
+	h.log.Debug("incubating htlc output")
 
-	err := h.IncubateOutputs(
+	// The utxo nursery will take care of broadcasting the second-level
+	// timeout tx and sweeping its output once it confirms.
+	return h.IncubateOutputs(
 		h.ChanPoint, fn.Some(h.htlcResolution),
 		fn.None[lnwallet.IncomingHtlcResolution](),
 		h.broadcastHeight, h.incomingHTLCExpiryHeight,
 	)
-	if err != nil {
-		return err
-	}
-
-	h.outputIncubating = true
```

Comment:
Yes! I also realized this after reviewing #9062. What's more, because the sweeper now handles re-offered inputs by first checking the db and mempool to grab the existing tx, it means it's safe to re-offer the same inputs during startup. And since deep down the nursery also uses the sweeper, it means we can easily make all the resolvers stateless.

Atm another usage of the `outputIncubating` is to decide whether we should resolve the stage one or stage two HTLC, which can also be changed to rely on chain notifications. So the changes would be something like,
1. subscribe the htlc tx spending at the very beginning when initializing the resolver
2. when `Resolve` it, if the htlc tx is already spent, subscribe to the output of the htlc tx
    - if the output is spent, we are done here
    - otherwise, wait for the spend to resolve the output spent.

Basically we just outsource the info provided by `outputIncubating` to chain notification, further simplify the resolvers here. The only part I'm not sure is for neutrino backend, since there's no mempool, it can only rely on the sweeper's store to decide whether an input is new or not, which means we may need to store more info there (currently only the txid).

I actually plan to do this after the `blockbeat` is in to limit the scope, rough idea is 1) refactor a bit to get rid of `outputIncubating`; 2) do a db migration to remove the `outputIncubating` (maybe not needed?); 3) completely remove the nursery! WDYT?

Reply:
> why the mempool is relevant here and could cause problems for neutrino.

because of this,
https://github.com/lightningnetwork/lnd/blob/fa309c9a0ef1449b5524ec53720430aa4491a3d2/sweep/sweeper.go#L1212